### PR TITLE
refactor: hide tgraph, cpool and streaming under new usable api

### PR DESCRIPTION
**Pull request by JoanFM**
Goals:

First conversation starter to have an `object` that can be used by community that hides the `complexity` from `TopologyGraph` and `ConnectionPool`.

Conditions of the design:

- [x] Simple interface. (__init__, stream (as send), and close)
- [x] Do not receive `args` but specific arguments mapping to concepts/features

Things to do:

- [ ] Proper concept naming
- [ ] Naming of arguments and methods
- [x] DocStrings
- [ ] Decision of interface
- [x] Refactor the request_generator part, do not import client code from server

### CODE REVIEW

**Review by JohannesMessner — COMMENTED** [review]
i like the direction of this! IMO this also makes the code architecture clearer for our internal purposes. I think we will need a more descriptive name for `GatewayBFF`, but can't think of something right now.

File: jina/serve/bff.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        )
+        for deployment_name, addresses in deployments_addresses.items():
+            for address in addresses:
+                connection_pool.add_connection(
+                    deployment=deployment_name, address=address, head=True
+                )
+
+        return connection_pool
+
+    def stream(self, *args, **kwargs):
```
Comment (JohannesMessner) [inline jina/serve/bff.py:93]: So to use this directly, people users would create Documents out of whatever data their receieve through their own API, and pass those to this, right?

File: jina/serve/bff.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    Wrapper object to be used in a BFF or in the Gateway. Naming to be defined
+    """
+
+    def __init__(
+            self,
+            graph_representation: Dict,
+            executor_addresses: Dict[str, Union[str, List[str]]],
+            graph_conditions: Dict,
+            deployments_disable_reduce: List[str],
+            timeout_send: Optional[float] = None,
```
[the remark this answers is not in the document]
Comment (JoanFM) [reply jina/serve/bff.py]: could be

File: jina/serve/bff.py (hunk 1 above)
Comment (JoanFM) [reply jina/serve/bff.py:93]: this is the one to serve how we already use. I think users should use the `stream_docs` one

[the remark this answers is not in the document]
Comment (JoanFM) [reply jina/serve/bff.py:93]: This we can make it private later. I do not want to expose this to the user. This is just to keep the old usage in our own gateways.

**Review by JohannesMessner — APPROVED** [review]

**Pull request merged**